runtime/common/src/mmr.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! A pallet responsible for creating Merkle Mountain Range (MMR) leaf for current block.
+
+use beefy_primitives::ValidatorSetId;
+use sp_core::H256;
+use sp_std::prelude::*;
+use frame_support::{decl_error, decl_module, decl_storage, RuntimeDebug,};
+use pallet_mmr::primitives::LeafDataProvider;
+use parity_scale_codec::{Encode, Decode};
+use runtime_parachains::paras;
+
+/// A leaf that get's added every block to the MMR constructed by [pallet_mmr].
+#[derive(RuntimeDebug, PartialEq, Eq, Clone, Encode, Decode)]
+pub struct MmrLeaf<Hash, MerkleRoot> {
+	/// Current block parent hash.
+	pub parent_hash: Hash,
+	/// A merkle root of all registered parachain heads.
+	pub parachain_heads: MerkleRoot,
+	/// A merkle root of the next beefy authority set.
+	pub beefy_next_authority_set: (ValidatorSetId, MerkleRoot),
+}
+
+impl<T: Config> LeafDataProvider for Module<T> where
```

Comment:
I think the placement of this block is a bit awkward. I'd move it further down the file where all the other `Module` impl blocks are